app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -740,8 +782,37 @@ public void toggleMoreResultsBox() {
     }
 
     @OnClick({R.id.inputAppraisalExpandBox})
+    /**
+     * Method called when user presses the text to expand the appraisal box on the input screen
+     */
     public void toggleAppraisalBox() {
         toggleVisibility(inputAppraisalExpandBox, appraisalBox);
+        moveOverlayUpOrDownToMatchAppraisalBox();
+    }
+
+    /**
+     * Moves the entire overlay up if the appraisal box is visible
+     */
+    private void moveOverlayUpOrDownToMatchAppraisalBox() {
+        if (windowManager == null) return; //do nothing if window is not initiated
+        if (infoLayout.getLayoutParams() == null) return;
+
+        WindowManager.LayoutParams newParams = (WindowManager.LayoutParams) infoLayout.getLayoutParams();
+        if (appraisalBox.getVisibility() == View.VISIBLE) {
+            newParams.gravity = Gravity.TOP;
+        }else{
+            newParams.gravity = Gravity.BOTTOM;
+        }
+        windowManager.updateViewLayout(infoLayout, newParams);
+    }
+
+    /**
+     * Moves the overlay down to the bottom of the screen
+     */
+    private void moveOverlayDown() {
+        WindowManager.LayoutParams newParams = (WindowManager.LayoutParams) infoLayout.getLayoutParams();
+        newParams.gravity = Gravity.BOTTOM;
+        windowManager.updateViewLayout(infoLayout, newParams);
     }
 
```

Comment:
moveOverlayDown is mostly duplicate code from moveOverlayUpOrDownToMatchAppraisalBox without the logic.

Maybe we can instead create a moveOverlayUp method and leave just the logic in moveOverlayUpOrDownToMatchAppraisalBox to avoid duplication calling the other two when needed.

Other way could be passing a parameter to "moveOverlay" true to "up" and false to "down" or something like that, since otherwise moveOverlayUp and moveOverlayDown would, once again, share most of the code.